django/utils/autoreload.py

```diff
@@ -236,15 +236,25 @@ def __init__(self):
 
     def watch_dir(self, path, glob):
         path = Path(path)
-        if not path.is_absolute():
-            raise ValueError('%s must be absolute.' % path)
+
+        try:
+            path = path.absolute()
+        except FileNotFoundError:
+            logger.debug('Skipping watching directory %s as it cannot be resolved.', path, exc_info=True)
+            return
+
         logger.debug('Watching dir %s with glob %s.', path, glob)
         self.directory_globs[path].add(glob)
 
     def watch_file(self, path):
         path = Path(path)
-        if not path.is_absolute():
-            raise ValueError('%s must be absolute.' % path)
+
+        try:
+            path = path.absolute()
+        except FileNotFoundError:
+            logger.debug('Skipping watching file %s as it cannot be resolved.', path, exc_info=True)
+            return
+
```

Comment:
Do we need these changes? :thinking: `Path.absolute()` raises `FileNotFoundError`, so why not catch it in `watch_for_translation_changes()`, e.g.
```python
for path in directories:
    try:
        absolute_path = path.absolute()
        sender.watch_dir(absolute_path, '**/*.mo')
    except FileNotFoundError:
        logger.debug('Skipping watching file %s as it cannot be resolved.', path, exc_info=True)

```

Reply:
I felt like this logic should belong in the reloader as it's a bit easier to test, and keeps it all centralized. I'm happy to do it like you suggest if you have strong feelings.

Reply:
I'm just afraid that we're changing a current behavior that works properly :thinking:, but I'm fine with your proposition. 

Off-topic: Do we need `watch_file()` at all? It looks unused :thinking: .

Reply:
@orf What do you think about removing `watch_file()`?

Reply:
See #11590

Reply:
Sorry for the late reply, yep, lets get rid of it 👍

Reply:
Great, thanks! I will merge #11590 and rebase this PR :+1: